src/lib/rateLimiter.ts

```
@@ -0,0 +1,37 @@
+import { RateLimiter } from './types.ts'
+
+export class TokenBucketRateLimiter implements RateLimiter {
+  private tokens: number
+  private lastRefill: number
+  private readonly maxTokens: number
+  private readonly refillRate: number
+
+  constructor(tokensPerSecond: number = 10) {
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)